plugins/record-hook/src/record.utils.ts

```diff
@@ -0,0 +1,117 @@
+import type { Flatfile } from '@flatfile/api'
+import { FlatfileRecord, FlatfileRecords } from '@flatfile/hooks'
+import type { FlatfileEvent } from '@flatfile/listener'
+import { logError, logInfo } from '@flatfile/util-common'
+import { RecordTranslator } from './record.translator'
+
+// deepEqual is a generic object comparison function that is ambivalent to the key order
+export function deepEqual(obj1, obj2) {
+  if (obj1 === obj2) {
+    return true
+  }
+
+  if (
+    typeof obj1 !== 'object' ||
+    obj1 === null ||
+    typeof obj2 !== 'object' ||
+    obj2 === null
+  ) {
+    return false
+  }
+
+  let keysObj1 = Object.keys(obj1)
+  let keysObj2 = Object.keys(obj2)
+
+  if (keysObj1.length !== keysObj2.length) {
+    return false
+  }
+
+  for (let key of keysObj1) {
+    if (!keysObj2.includes(key) || !deepEqual(obj1[key], obj2[key])) {
+      return false
+    }
+  }
+
+  return true
+}
+
+// BACKGROUND:
```

Comment:
Awesome description! Wonder if we should add this to the ts-doc for this function somehow and bubble it up so customers can stumble upon it

Reply:
This started as a note-to-self, definitely more than that now. Adding this to the ts-doc makes sense.